basis_universal: Rediff patches, remove Windows encoding fix

akien-mga opened the pull request:
Convert the file to UTF-8 properly instead.

akien-mga committed:
basis_universal: Rediff patches, remove Windows encoding fix

Convert the file to UTF-8 properly instead.

DanielKinsman commented:
https://github.com/BinomialLLC/basis_universal/blob/v1_60/encoder/basisu_enc.cpp is also different from ours and the difference is not in the patches at all. Might be others. Perhaps best to start with a fresh copy of everything from basis universal?

akien-mga commented:
I started from a fresh copy of `v1_60`. What difference do you see?

akien-mga commented:
This is the only difference we have to `v1_60`:

```
~/.../thirdparty/basis_universal (basis_universal-rediff-patches %)$ rm encoder/ transcoder/ -rf
~/.../thirdparty/basis_universal (basis_universal-rediff-patches *%)$ tar xf ~/Downloads/basis_universal-1_60.tar.gz 
~/.../thirdparty/basis_universal (basis_universal-rediff-patches *%)$ cp -r basis_universal-1_60/encoder basis_universal-1_60/transcoder/ basis_universal-1_60/LICENSE .
~/.../thirdparty/basis_universal (basis_universal-rediff-patches *%)$ rm -rf encoder/{jpgd.{cpp,h},3rdparty/{qoi.h,tinydds.h,tinyexr.cpp,tinyexr.h}}
~/.../thirdparty/basis_universal (basis_universal-rediff-patches *%)$ git status
On branch basis_universal-rediff-patches
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   encoder/basisu_astc_hdr_6x6_enc.h
        modified:   encoder/basisu_comp.cpp
        modified:   encoder/basisu_enc.cpp
        modified:   encoder/basisu_gpu_texture.cpp
        modified:   transcoder/basisu_containers.h
        modified:   transcoder/basisu_transcoder.cpp

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        basis_universal-1_60/

no changes added to commit (use "git add" and/or "git commit -a")
~/.../thirdparty/basis_universal (basis_universal-rediff-patches *%)$ git add encoder/ transcoder/
~/.../thirdparty/basis_universal (basis_universal-rediff-patches +%)$ for n in {1..5}; do patch -p3 < patches/000${n}*; done
patching file encoder/basisu_comp.cpp
patching file transcoder/basisu_transcoder.cpp
patching file encoder/basisu_enc.cpp
patching file encoder/basisu_enc.cpp
patching file encoder/basisu_enc.cpp
patching file encoder/basisu_gpu_texture.cpp
patching file transcoder/basisu_containers.h
~/.../thirdparty/basis_universal (basis_universal-rediff-patches *+%)$ git status
On branch basis_universal-rediff-patches
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        modified:   encoder/basisu_astc_hdr_6x6_enc.h
        modified:   encoder/basisu_comp.cpp
        modified:   encoder/basisu_enc.cpp
        modified:   encoder/basisu_gpu_texture.cpp
        modified:   transcoder/basisu_containers.h
        modified:   transcoder/basisu_transcoder.cpp

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   encoder/basisu_comp.cpp
        modified:   encoder/basisu_enc.cpp
        modified:   encoder/basisu_gpu_texture.cpp
        modified:   transcoder/basisu_containers.h
        modified:   transcoder/basisu_transcoder.cpp

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        basis_universal-1_60/

~/.../thirdparty/basis_universal (basis_universal-rediff-patches *+%)$ git add encoder/ transcoder/
~/.../thirdparty/basis_universal (basis_universal-rediff-patches +%)$ git status
On branch basis_universal-rediff-patches
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        modified:   encoder/basisu_astc_hdr_6x6_enc.h

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        basis_universal-1_60/

~/.../thirdparty/basis_universal (basis_universal-rediff-patches +%)$ git diff --staged
diff --git a/thirdparty/basis_universal/encoder/basisu_astc_hdr_6x6_enc.h b/thirdparty/basis_universal/encoder/basisu_astc_hdr_6x6_enc.h
old mode 100755
new mode 100644
index 8583ddcc7b..0d6d2ae936
--- a/thirdparty/basis_universal/encoder/basisu_astc_hdr_6x6_enc.h
+++ b/thirdparty/basis_universal/encoder/basisu_astc_hdr_6x6_enc.h
@@ -16,7 +16,7 @@ namespace astc_6x6_hdr
        {
                // Important: The Delta ITP colorspace error metric we use internally makes several assumptions about the nature of the HDR RGB inputs supplied to the encoder.
                // This encoder computes colorspace error in the ICtCp (or more accurately the delta ITP, where CT is scaled by .5 vs. ICtCp to become T) colorspace, so getting this correct is important.
-               // By default the encoder assumes the input is in absolute luminance (in nits or candela per square meter, cd/m²), specified as positive-only linear light RGB, using the REC 709 colorspace gamut (but NOT the sRGB transfer function, i.e. linear light).
+               // By default the encoder assumes the input is in absolute luminance (in nits or candela per square meter, cd/m<B2>), specified as positive-only linear light RGB, using the REC 709 colorspace gamut (but NOT the sRGB transfer function, i.e. linear light).
                // If the m_rec2020_bt2100_color_gamut flag is true, the input colorspace is treated as REC 2020/BT.2100 (which is wider than 709).
                // For SDR/LDR->HDR upconversion, the REC 709 sRGB input should be converted to linear light (sRGB->linear) and the resulting normalized linear RGB values scaled by either 80 or 100 nits (the luminance of a typical SDR monitor). 
                // SDR upconversion to normalized [0,1] (i.e. non-absolute) luminances may work but is not supported because ITP errors will not be predicted correctly.
```

Repiteo commented:
Thanks!
